carbon/app.py

```diff
@@ -240,3 +263,36 @@ def run_connection_test(self) -> None:
         else:
             logger.info("Successfully connected to the Symplectic Elements FTP server")
             ftps.quit()
+
+
+def run_all_connection_tests(
+    engine: DatabaseEngine, pipe: DatabaseToFilePipe | DatabaseToFtpPipe
+) -> None:
+    """Run connection tests for the Data Warehouse and Elements FTP server.
+
+    Args:
+        engine (DatabaseEngine): A configured carbon.database.DatabaseEngine that can
+            connect to the Data Warehouse.
+        pipe (DatabaseToFilePipe | DatabaseToFtpPipe): The pipe used to run the
+            data feed. If the pipe is an instance of carbon.app.DatabaseToFtpPipe,
+            a connection test for the Elements FTP server is run.
+    """
+    # test connection to the Data Warehouse
+    try:
+        engine.run_connection_test()
+    except Exception:  # noqa: BLE001
+        logger.error(  # noqa: TRY400
```

Comment:
Why not use `logger.exception` here to get the traceback as well?

Reply:
The reason for this is that the traceback would already be logged at the lower level. 

For `engine.run_connection_test()` the try-except code blocks in lines 113-122 of `carbon.database` make it such that the traceback will be logged for all other errors _except_ `DatabaseError`, which are pretty straight forward. 

This [log](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:log-groups/log-group/carbon-ecs-stage/log-events/carbon$252Fcarbon-ecs-stage$252Ffccf51066550475cbca712465f035304) shows what it would be like if the Data Warehouse connection test failed and `logger.exception` is used at the higher-level (in `carbon.app`). 

The error message is summarized by the fifth line of the log: 
<img width="1408" alt="image" src="https://github.com/MITLibraries/carbon/assets/29102054/e2f31b93-cff6-47a8-b893-9c8e3c760fc8">

If we use `logger.exception`, the log ends with a super long traceback that ultimatley shows the same `DatabaseError` message--rather than a clean message indicating the program is exiting.

---

In short, this code allows us to throw exceptions with traceback at the low-level and cleanly exit the program (i.e. if any of the connection tests fail, the logs will end with: 

>  ERROR carbon.app.run_all_connection_tests(): The Data Warehouse connection test failed. The application is exiting.

Any traceback for non-`DatabaseError` exceptions would appear _before_ this exit message.

---

The same reason applies for your question below! Let me know if you need further clarification.

Reply:
The current code produces a much [cleaner, shorter log](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:log-groups/log-group/carbon-ecs-stage/log-events/carbon$252Fcarbon-ecs-stage$252F37b73d41b7074437b0618a2e2e56d457
) (in the case of `DatabaseError` exceptions): 
<img width="1410" alt="image" src="https://github.com/MITLibraries/carbon/assets/29102054/10cd8e3b-d0f3-4318-b6c3-83672b01378a">

Reply:
Got it, I appreciate your clarification!